implementation/endpoints/src/tcp_client_endpoint_impl.cpp

```diff
@@ -178,7 +178,7 @@ void tcp_client_endpoint_impl::connect() {
         std::string its_device(configuration_->get_device());
         if (its_device != "") {
             if (setsockopt(socket_->native_handle(),
-                    SOL_SOCKET, SO_BINDTODEVICE, its_device.c_str(), (socklen_t)its_device.size()) == -1) {
+                    SOL_SOCKET, SO_BINDTODEVICE, its_device.c_str(), socklen_t(its_device.size())) == -1) {
```

Comment:
Avoid C-style casts. Prefer using named casts, instead.

```suggestion
                    SOL_SOCKET, SO_BINDTODEVICE, its_device.c_str(), static_cast<socklen_t>(its_device.size())) == -1) {
```

Reply:
This isn't a c-style cast, the change here is changing from a c-style cast to a proper [initialization (3)](https://en.cppreference.com/w/cpp/language/initialization)

Reply:
Correct me if I'm wrong, but `socklen_t` is just an alias for an integer of at least 32-bits. Since we're converting from a size_t to an u32 (or greater), I think a `static_cast` is the best option. Don't you?

Reply:
In general I prefer to avoid `static_cast` when I can.  It's safe here because we're going to a bigger type but that's not always obvious.  To avoid potential narrowing, I think it's usually safer to use initialization for the cast.

I probably should have used curly braces.

When going to a smaller type on the other-hand, then I prefer to use [gsl::narrow](https://github.com/microsoft/GSL/blob/main/include/gsl/narrow) to safely cast - unless I'm confident it's always safe and then I use `gsl::narrow_cast` or `static_cast`.

I can change it to anything here, I think initialization is better, but anything here would be better than a c-cast. :)

Reply:
You make a fair point. I would just request that you use `{}`-initialization, instead of `()`-initialization.